script/path.js

```diff
@@ -0,0 +1,361 @@
+		} else if(store.type == 'tool' && store.desc != "undefined") {
+			$('<div>').addClass('row_key').text(store.desc).appendTo(tt);
+		}
+
+		$('<div>').addClass('row_key').text(_('weight')).appendTo(tt);
+		$('<div>').addClass('row_val').text(Path.getWeight(key)).appendTo(tt);
+		$('<div>').addClass('row_key').text(_('available')).appendTo(tt);
+		$('<div>').addClass('row_val').addClass('numAvailable').text(numAvailable).appendTo(tt);
+		
+		return row;
+	},
+	
+	increaseSupply: function(btn) {
+		var supply = $(this).closest('.outfitRow').attr('key');
+		Engine.log('increasing ' + supply + ' by up to ' + btn.data);
+		var cur = Path.outfit[supply];
+		cur = typeof cur == 'number' ? cur : 0;
+		if(Path.getFreeSpace() >= Path.getWeight(supply) && cur < $SM.get('stores["'+supply+'"]', true)) {
+			var maxExtraByWeight = Math.floor(Path.getFreeSpace() / Path.getWeight(supply));
+			var maxExtraByStore  = $SM.get('stores["'+supply+'"]', true) - cur;
+			Path.outfit[supply] = cur + Math.min(btn.data, maxExtraByWeight, maxExtraByStore);
+			$SM.set('outfit['+supply+']', Path.outfit[supply]);
+			Path.updateOutfitting();
+		}
+	},
+	
+	decreaseSupply: function(btn) {
+		var supply = $(this).closest('.outfitRow').attr('key');
+		Engine.log('decreasing ' + supply + ' by up to ' + btn.data);
+		var cur = Path.outfit[supply];
+		cur = typeof cur == 'number' ? cur : 0;
+		if(cur > 0) {
+			Path.outfit[supply] = Math.max(0, cur - btn.data);
+			$SM.set('outfit['+supply+']', Path.outfit[supply]);
+			Path.updateOutfitting();
+		}
+	},
+	
+	onArrival: function(transition_diff) {
+		Path.setTitle();
+		Path.updateOutfitting();
+		Path.updatePerks(true);
+
+		Engine.moveStoresView($('#perks'), transition_diff);
+	},
+	
+	setTitle: function() {
+		document.title = _('A Dusty Path');
+	},
+	
+	embark: function() {
+		for(var k in Path.outfit) {
+			$SM.add('stores["'+k+'"]', -Path.outfit[k]);
+		}
+		World.onArrival();
+		$('#outerSlider').animate({left: '-700px'}, 300);
+		Engine.activeModule = World;
+	},
+	
+	handleStateUpdates: function(e){
+		if(e.category == 'character' && e.stateName.indexOf('character.perks') === 0 && Engine.activeModule == Path){
+			Path.updatePerks();
+		} else if(e.category == 'income' && Engine.activeModule == Path){
+			Path.updateOutfitting();
+		}
+	},
+
+	scrollSidebar: function(direction, reset){
+
+		if( typeof reset != "undefined" ){
+			$('#perks').css('top', '0px');
+			$('#storesContainer').css('top', '206px');
+			Path._STORES_OFFSET = 0;
+			return;
+		}
+		
+		var momentum = 10;
+
+		if( direction == 'up' )
+			momentum = momentum * -1;
+
+		if( direction == 'down' && inView( direction, $('#perks') ) ){
+
+			return false;
+
+		}else if( direction == 'up' && inView( direction, $('#storesContainer') ) ){
+
+			return false;
+
+		}
+
+		scrollByX( $('#perks'), momentum );
+		scrollByX( $('#storesContainer'), momentum );
+		Path._STORES_OFFSET += momentum;
+
+	}
+};
```

Comment:
This class as a whole is fairly large and potentially a bloater. Some methods should potentially be split into smaller subclasses. For instance, methods dealing with armor/outfitting, perks, and free space could all be made into their own class, which would make it so this class does not suffer from being concerned with far too many responsibilities.